client/src/components/ExpandedRowComponent.tsx

```diff
@@ -0,0 +1,128 @@
+        };
+        const showClaim = !user || !data.reviewers?.includes(user?.displayName);
+        const showApprove = !user || !data.approvers?.includes(user?.displayName);
+        const showChange = !user || !data.requestChanges?.includes(user?.displayName);
+        const [editing, setEditing] = useState(false);
+        const [errorMessage, setErrorMessage] = useState(null as null | string);
+        const [saveClicked, setSaveClicked] = useState(false);
+        const dirtyData = {...data};
+
+        const handleSaveClicked = ({currentTarget}: MouseEvent<HTMLButtonElement>) => {
+            if (JSON.stringify(dirtyData) !== JSON.stringify(data)) {
+                const value = currentTarget.getAttribute('value');
+
+                setSaveClicked(true);
+                fetchData(`/api/action/save/${value}`, {
+                    method: 'POST',
+                    headers: {
+                        'Content-Type': 'application/json'
+                    },
+                    body: JSON.stringify({
+                        note: dirtyData.note,
+                        jiraTicket: dirtyData.jiraTicket,
+                    })
+                })
+                    .then((result) => {
+                        if (!result.data) {
+                            setErrorMessage('Error saving data, see console error log for details.');
+                            logError(`Error saving: /api/action/save/${value}`, result);
+                            return;
+                        }
+                        Object.assign(data, result.data);
+                        onUpdate && onUpdate({
+                            data: result.data,
+                            action: 'save',
+                        });
+                    })
+                    .catch((e) => {
+                        setErrorMessage('Error saving data, see console error log for details.');
+                        logError(`Error sending: /api/action/save/${value}`, e);
+                    })
+                    .finally(() => setEditing(false));
+
+            } else {
+                setEditing(false);
+            }
+        }
+
+        return (
+            <div style={{paddingLeft: '3em'}} className="expanded-row">
+                {!editing && (
+                    <div>
+                        <div className="expanded-nav">
+                            <div className="left">
+                                {showClaim && (<ActionButton label="Claim" reviewId={data.id}/>)}
+                                {showChange && (<ActionButton label="Request Change" reviewId={data.id}/>)}
+                                {showApprove && (<ActionButton label="Approve" reviewId={data.id}/>)}
+                                <ActionButton label="Close" reviewId={data.id}/>
+                                <ActionButton label="Remove" reviewId={data.id}/>
+                                <button onClick={() => setEditing(true)}>Edit</button>
```

Comment:
For a11y every control should have a unique name - can use a visually hidden class to say "review #{$data.id}` or something like that